src/components/results/common/global-filter/results-global-filter.tsx

```diff
@@ -0,0 +1,397 @@
+import {
+    ElementType,
+    TreeViewFinderNodeProps,
+    mergeSx,
+    DirectoryItemSelector,
+    fetchElementsInfos,
+    ElementAttributes,
+    fetchDirectoryElementPath,
+    useSnackMessage,
+} from '@gridsuite/commons-ui';
+import { EQUIPMENT_TYPES } from '../../../utils/equipment-types';
+import { UUID } from 'crypto';
+import { GlobalFilter } from './global-filter-types';
+import { getResultsGlobalFiltersChipStyle, resultsGlobalFilterStyles } from './global-filter-styles';
+import SelectableGlobalFilters from './selectable-global-filters';
+import Tooltip from '@mui/material/Tooltip';
+import IconButton from '@mui/material/IconButton';
+import { computeFullPath } from '../../../../utils/compute-title';
+
+const recentFilter: string = 'recent';
+
+export interface ResultsGlobalFilterProps {
+    onChange: (filters: GlobalFilter[]) => void;
+    filterableEquipmentTypes: EQUIPMENT_TYPES[];
+    filters: GlobalFilter[];
+}
+
+const emptyArray: GlobalFilter[] = [];
+const DEFAULT_NB_OPTIONS_DISPLAYED: number = 10;
+
+const ResultsGlobalFilter: FunctionComponent<ResultsGlobalFilterProps> = ({
+    onChange,
+    filterableEquipmentTypes,
+    filters = emptyArray,
+}) => {
+    const intl = useIntl();
+    const { snackError } = useSnackMessage();
+    const { translate } = useLocalizedCountries();
+    const dispatch = useDispatch<AppDispatch>();
+    const recentGlobalFilters: GlobalFilter[] = useSelector((state: AppState) => state.recentGlobalFilters);
+    // Map <FilterType, number of options of this type>
+    // -1 number of options means that the user required everything to be displayed no matter the number of options
+    const [numberOfOptions, setNumberOfOptions] = useState<Map<string, number>>(
+        new Map([
+            [FilterType.COUNTRY, 0],
+            [FilterType.VOLTAGE_LEVEL, 0],
+            [FilterType.FILTER, 0],
+        ])
+    );
+    const [directoryItemSelectorOpen, setDirectoryItemSelectorOpen] = useState(false);
+    const [selectedGlobalFilters, setSelectedGlobalFilters] = useState<GlobalFilter[]>([]);
+
+    const getOptionLabel = useCallback(
+        (option: GlobalFilter) => {
+            switch (option.filterType) {
+                case FilterType.COUNTRY:
+                    return translate(option.label);
+                case FilterType.VOLTAGE_LEVEL:
+                    return option.label + ' kV';
+                case FilterType.FILTER:
+                    return option.label;
+            }
+        },
+        [translate]
+    );
+
+    const handleChange = useCallback(
+        (globalFilters: GlobalFilter[]): void => {
+            let globalFiltersToAddToRecents: GlobalFilter[] = globalFilters;
+            Promise.all(
+                // checks if the generic filters still exist, and sets their path value
+                globalFilters
+                    .filter((globalFilter: GlobalFilter) => globalFilter.filterType === FilterType.FILTER)
```

Comment:
to avoid test in map :

```suggestion
                    .filter((globalFilter: GlobalFilter) => globalFilter.filterType === FilterType.FILTER && globalFilter.uuid)
```

Reply:
I agree that it would be better but I put the test here because of SonarLint checks : it doesn't detect the test and displays an error on `fetchDirectoryElementPath(fetchedGlobalFilter.uuid)`

I am going to push it anyway to see if it is mergeable.

Edit : refused. I have to keep the "bad" version of the code.